File: admin/src/test/java/com/beat/admin/promotion/api/response/CarouselPresignedUrlFindAllResponseTest.java
package com.beat.admin.promotion.api.response;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.Map;

import org.junit.jupiter.api.Test;

import com.beat.admin.promotion.application.result.AdminPromotionPresignedUrlResults.CarouselPresignedUrlsResult;
import com.beat.contracts.storage.CarouselPresignedUpload;

class CarouselPresignedUrlFindAllResponseTest {

	@Test
	void responsePreservesLegacyUrlsAndAddsExplicitUploadMetadata() {
		CarouselPresignedUrlFindAllResponse response = new CarouselPresignedUrlFindAllResponse(
			new CarouselPresignedUrlsResult(Map.of("carousel.png",
				CarouselPresignedUpload.of("signed-upload-url", "dev/carousel/carousel.png")))
		);

		assertEquals(Map.of("carousel.png", "signed-upload-url"), response.getCarouselPresignedUrls());
		assertEquals("signed-upload-url", response.getCarouselPresignedUploads().get("carousel.png").getUploadUrl());
		assertEquals("dev/carousel/carousel.png", response.getCarouselPresignedUploads().get("carousel.png").getImageKey());
	}
}

File: admin/src/test/kotlin/com/beat/admin/promotion/api/response/CarouselPresignedUrlFindAllResponseTest.kt
package com.beat.admin.promotion.api.response

import com.beat.admin.promotion.application.result.AdminPromotionPresignedUrlResults.CarouselPresignedUrlsResult
import com.beat.contracts.storage.CarouselPresignedUpload
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class CarouselPresignedUrlFindAllResponseTest {

    @Test
    fun responsePreservesLegacyUrlsAndAddsExplicitUploadMetadata() {
        val response = CarouselPresignedUrlFindAllResponse(
            CarouselPresignedUrlsResult(
                mapOf("carousel.png" to CarouselPresignedUpload.of("signed-upload-url", "dev/carousel/carousel.png")),
            ),
        )

        assertEquals(mapOf("carousel.png" to "signed-upload-url"), response.carouselPresignedUrls)
        assertEquals("signed-upload-url", response.carouselPresignedUploads["carousel.png"]!!.uploadUrl)
        assertEquals("dev/carousel/carousel.png", response.carouselPresignedUploads["carousel.png"]!!.imageKey)
    }
}